File: src/ChatApp/Services/user-service/src/main/java/com/chatapp/userservice/entity/File.java
package com.chatapp.userservice.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.sql.Timestamp;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
@Table(name = "files")
public class File {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id;

    @Column(name = "name", columnDefinition = "VARCHAR(255)", nullable = false)
    String name;

    @Column(name = "url", columnDefinition = "VARCHAR(255)", nullable = false)
    String url;

    @Column(name = "created_at", columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP", nullable = false)
    Timestamp createdAt;

    @ManyToOne()
    @JoinColumn(name = "owner_id")
    User user;

    @ManyToOne()
    @JoinColumn(name = "room_id")
    Room room;
}

File: src/ChatApp/Services/Message/ChatApp.Message/Models/File.cs
using System;
using System.Collections.Generic;

namespace ChatApp.Message.Models;

public partial class File
{
    public int Id { get; set; }

    public DateTime CreatedAt { get; set; }

    public string Name { get; set; } = null!;

    public string Url { get; set; } = null!;

    public int? MessageId { get; set; }

    public int? RoomId { get; set; }

    public int? OwnerId { get; set; }

    public virtual Message? Message { get; set; }

    public virtual User? Owner { get; set; }

    public virtual Room? Room { get; set; }
}
